src/components/radio-group/daikin-radio-group.ts

```diff
@@ -0,0 +1,221 @@
+import { LitElement, css, html, unsafeCSS, type PropertyValues } from "lit";
+import {
+  customElement,
+  property,
+  queryAssignedElements,
+} from "lit/decorators.js";
+
+import { cva } from "class-variance-authority";
+import tailwindStyles from "../../tailwind.css?inline";
+import type { MergeVariantProps } from "../../type-utils";
+import { DaikinRadio } from "../radio/daikin-radio";
+
+const radioGroupCN = cva(["flex", "gap-2", "py-2", "pr-2"], {
+  variants: {
+    orientation: {
+      horizontal: ["flex-col"],
+      vertical: ["flex-row", "gap-3"],
+    },
+  },
+});
+
+type RadioGroupProps = MergeVariantProps<typeof radioGroupCN>;
+
+/**
+ * Radio groups are used to group multiple radio buttons so that make sure that only one will be selected in the group
+ *
+ * Hierarchies:
+ * - `daikin-radio-group` > `daikin-radio`
+ * - `daikin-input-group` > `daikin-radio-group` > `daikin-radio`
+ *
+ * @fires change - A custom event emitted when current checked radio changed.
+ *
+ * @slot - A slot for radio buttons. Place `daikin-radio` elements here.
+ *
+ * @example
+ *
+ * ```html
+ * <daikin-radio-group>
+ *   <daikin-radio name="name" value="value1" label="Option1"></daikin-radio>
+ *   <daikin-radio name="name" value="value2" label="Option2"></daikin-radio>
+ *   <daikin-radio name="name" value="value3" label="Option3"></daikin-radio>
+ * </daikin-radio-group>
+ * ```
+ *
+ * ```html
+ * <daikin-input-group label="Label text" helper="Helper text">
+ *   <daikin-radio-group>
+ *     <daikin-radio name="name" value="value1" label="Option1"></daikin-radio>
+ *     <daikin-radio name="name" value="value2" label="Option2"></daikin-radio>
+ *     <daikin-radio name="name" value="value3" label="Option3"></daikin-radio>
+ *   </daikin-radio-group>
+ * </daikin-input-group>
+ * ```
+ */
+@customElement("daikin-radio-group")
+export class DaikinRadioGroup extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: inline-block;
+    }
+  `;
+
+  @queryAssignedElements({ selector: "daikin-radio" })
+  private readonly _radios!: Array<DaikinRadio>;
+
+  /**
+   * Specify the radio group orientation
+   */
+  @property({ type: String })
+  orientation: RadioGroupProps["orientation"] = "horizontal";
+
+  /**
+   * The form name.
+   */
+  @property({ type: String, reflect: true })
+  name = "";
+
+  /**
+   * `value` of the currently selected radio.
+   * see {@link DaikinRadio.value}
+   */
+  @property({ type: String, reflect: true })
+  value = "";
+
+  private _updateRadioIndex(daikinRadio: DaikinRadio, index: string) {
+    const input = daikinRadio.shadowRoot?.querySelector("input");
+    input?.setAttribute("tabindex", index);
+  }
+
+  private _updateRadios(value: string, name: string) {
```

Comment:
value and name should always reference `this.value` and `this.name`.

```suggestion
  private _updateRadios() {
```